src/OpenTracing.MockTracer/MockTracer.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using OpenTracing.Propagation;
+
+namespace OpenTracing.MockTracer
+{
+    /// <summary>
+    /// In-memory <see cref="ITracer"/> implementation designed to make it
+    /// easy to test assertions against the OpenMetrics API and its use
+    /// inside frameworks.
+    /// </summary>
+    public class MockTracer : ITracer
+    {
+        private ConcurrentQueue<MockSpan> _finishedSpans = new ConcurrentQueue<MockSpan>();
+        private readonly IPropagator _propagator;
+
+        public static readonly TextPropagator TextMapPropagator = new TextPropagator();
+
+        public MockTracer() : this(TextMapPropagator)
+        {
+        }
+
+        public MockTracer(IPropagator propagator)
+        {
+            _propagator = propagator;
+        }
+
+        public void FinishSpan(MockSpan span)
+        {
+            _finishedSpans.Enqueue(span);
+        }
+
+        public IEnumerable<MockSpan> FinishedSpans => _finishedSpans;
```

Comment:
Java does

```Java
@return a copy of all finish()ed MockSpans started by this MockTracer (since construction or the last call to
     * MockTracer.reset()).
```

However by using a lazily evaluated IEnumerable rather than an eagerly evaluated List, the following code would not do what is expected

```C#
// pseudo-code
var tracer = new MockTracer(...);
ISpan a = ...;
tracer.FinishSpan(a);
var justA = tracer.FinishedSpans;
ISpan b = ...;
tracer.FinishSpan(b);
var aAndB= tracer.FinishSpans;

// At this point, justA == aAndB and both have both a and b
```